cdap-security/src/test/java/co/cask/cdap/security/authorization/InMemoryAuthorizer.java

```diff
@@ -168,24 +170,32 @@ public void removeRoleFromPrincipal(Role role, Principal principal) throws RoleN
         privileges.addAll(getPrivileges(role));
```

Comment:
I think the purpose of using listRoles here was to filter roles for a particular principal. We would have to add the same logic here too, if we looped over the roleToPrincipals map